app/src/main/java/com/celzero/bravedns/util/OrbotHelper.kt

```diff
@@ -286,10 +291,35 @@ class OrbotHelper(private val persistentState: PersistentState, private val prox
         return builder
     }
 
+    private fun fetchColor(context : Context): Int {
+        return when (persistentState.theme) {
+            Constants.THEME_SYSTEM_DEFAULT -> {
+                if (isDarkThemeOn(context)) {
+                    R.color.accentGoodBlack
+                } else {
+                    R.color.negative_white
+                }
+            }
+            Constants.THEME_LIGHT -> {
+                R.color.negative_white
+            }
+            Constants.THEME_DARK -> {
+                R.color.accent_good
+            }
+            else -> {
+                R.color.accentGoodBlack
+            }
+        }
+    }
+
+    private fun isDarkThemeOn(context : Context): Boolean {
+        return context.resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK == Configuration.UI_MODE_NIGHT_YES
+    }
```

Comment:
do not repeat yourself. do not copy-paste code around. find a way to abstract it out and use as approp.

Reply:
Removed `isDarkThemeOn` from OrbotHelper.
Placed in utilities.